src/services/core/model/src/main/java/org/geoserver/geofence/core/model/util/PwEncoder.java

```diff
@@ -21,9 +21,14 @@
  */
 public class PwEncoder {
 
-    // 123456789 123456789 123456789 12
-    private static final byte[] KEY = "installation dependant key needed".substring(0, 16)
-            .getBytes();
+    private static final byte[] KEY;
+    static {
+      String strKey = System.getProperty("GEOFENCE_PWENCODER_KEY");
+      if (strKey == null) {
+        strKey = "installation dependant key needed";
+      }
+      KEY = strKey.substring(0, 16).getBytes();
```

Comment:
This could throw an `IndexOutOfBoundsException` if  **strKey** is set from the System property, and the property value is not at least 16 characters long.